lib/active_model/serializer/utils.rb

```diff
@@ -0,0 +1,25 @@
+module ActiveModel::Serializer::Utils
+  module_function
+
+  # Gives the existing lookup paths that are suffixes of the fully qualified path of `klass`.
+  # @param [Object] klass
+  # @return [Array<Symbol>]
+  def nested_lookup_paths(klass)
+    paths = klass.to_s.split('::').reverse.reduce([]) { |a, e| a + [[e] + Array(a.last)] }.reverse
```

Comment:
So that as the test says

`Utils.nested_lookup_paths(TweetSerializer::ShareSerializer::AuthorSerializer) #=> [::TweetSerializer::ShareSerializer::AuthorSerializer,
                      ::ShareSerializer::AuthorSerializer,
                      ::AuthorSerializer,
                      ::Object]`